api/infrastructure/parallelcluster-api.yaml

```diff
@@ -302,8 +311,9 @@ Resources:
             Principal:
               AWS:
                 - !Sub arn:${AWS::Partition}:iam::${AWS::AccountId}:root
+      PermissionsBoundary: !If [HasPermissionBoundary, !Ref PermissionsBoundaryPolicy, !Ref 'AWS::NoValue']
       Policies:
-        - PolicyName: ParallelClusterApiInvokePolicy
+        - PolicyName: !Sub ${IAMRoleAndPolicyPrefix}ParallelClusterApiInvokePolicy
```

Comment:
You renamed the role above from `ParallelClusterApiUserRole`  to `PclusterApiUserRole`.
I suggest to be consistent in this renaming and align also the other roles.